generator/generator.go

```diff
@@ -35,7 +35,7 @@ func mutateHook(cfg *config.Config, usedTypes map[string]bool) func(b *modelgen.
 		// For more info see https://github.com/99designs/gqlgen/blob/master/docs/content/recipes/modelgen-hook.md
 		for _, model := range build.Models {
 			// only handle input type model
-			if schemaModel, ok := cfg.GQLConfig.Schema.Types[model.Name]; ok && (schemaModel.IsInputType() || cfg.Generate.ShouldOmitEmptyTypes()) {
+			if schemaModel, ok := cfg.GQLConfig.Schema.Types[model.Name]; ok && cfg.Generate.ShouldOmitEmptyTypes() {
```

Comment:
generate omitempty by gqlgen instead of mutateHook.

```go
	cfg.GQLConfig = &config.Config{
		EnableModelJsonOmitemptyTag:    cfg.Generate.EnableClientJsonOmitemptyTag,
	}
```